### DISCUSSION TITLE
Get pending message per subject in stream

### DISCUSSION THREAD

**Discussion by suchen-sci** [nats-io/nats-server#5216] [Q&A]
Hi team,

I'm curious if there's a method to obtain information about all pending messages per subject within a stream. 

For instance, if I have a stream with subjects such as `sub.123`, `sub.234`, `sub.345` and so on (maybe 10k+ subjects in a stream), and each subject has zero or one or more durable consumers (all with same durable name). 

How can I determine the number of pending messages for each subject? This information would help me decide if I need to create additional consumers for any particular subject. Some of the subject may have zero consumer since there are no messages there for them to consume.

Best wishes!

**Comment by neilalexander** [comment]
Instead of trying to query for pending info, which may end up being expensive if done repeatedly, you may instead want to consider a pattern whereby you have a separate headers-only consumer that subscribes to `sub.>` and then use those events to create more durable consumers that are subject-filtered if needed.
